src/stacks-hierarchy/navigation-types.ts

```diff
@@ -58,43 +58,40 @@ type FareContractDetailsRouteParams = {
   orderId: string;
 };
 
+type AfterLoginScreenType =
+  | NextScreenParams<'Root_TabNavigatorStack'>
+  | NextScreenParams<'Root_PurchaseOverviewScreen'>
+  | NextScreenParams<'Root_PurchaseConfirmationScreen'>
+  | NextScreenParams<'Root_ActiveTokenOnPhoneRequiredForFareProductScreen'>;
+
 export type Root_LoginActiveFareContractWarningScreenParams = {
-  afterLogin?:
-    | AfterLoginParams<'Root_TabNavigatorStack'>
-    | AfterLoginParams<'Root_PurchaseOverviewScreen'>
-    | AfterLoginParams<'Root_PurchaseConfirmationScreen'>;
+  afterLogin?: AfterLoginScreenType;
 };
 
 export type Root_LoginOptionsScreenParams = {
-  afterLogin?:
-    | AfterLoginParams<'Root_TabNavigatorStack'>
-    | AfterLoginParams<'Root_PurchaseOverviewScreen'>
-    | AfterLoginParams<'Root_PurchaseConfirmationScreen'>;
+  afterLogin?: AfterLoginScreenType;
 };
 
 export type Root_LoginPhoneInputScreenParams = {
-  afterLogin?:
-    | AfterLoginParams<'Root_TabNavigatorStack'>
-    | AfterLoginParams<'Root_PurchaseOverviewScreen'>
-    | AfterLoginParams<'Root_PurchaseConfirmationScreen'>;
+  afterLogin?: AfterLoginScreenType;
 };
 
 export type Root_LoginConfirmCodeScreenParams = {
   phoneNumber: string;
-  afterLogin?:
-    | AfterLoginParams<'Root_TabNavigatorStack'>
-    | AfterLoginParams<'Root_PurchaseOverviewScreen'>
-    | AfterLoginParams<'Root_PurchaseConfirmationScreen'>;
+  afterLogin?: AfterLoginScreenType;
 };
 
 export type Root_LoginRequiredForFareProductScreenParams = {
-  afterLogin?:
-    | AfterLoginParams<'Root_TabNavigatorStack'>
-    | AfterLoginParams<'Root_PurchaseOverviewScreen'>
-    | AfterLoginParams<'Root_PurchaseConfirmationScreen'>;
+  afterLogin?: AfterLoginScreenType;
   fareProductTypeConfig: FareProductTypeConfig;
 };
 
+export type Root_ActiveTokenOnPhoneRequiredForFareProductScreenParams = {
+  afterEnabled?:
```

Comment:
With the rename to `NextScreenParams` I think the parameter could also be `nextScreen`, and that we start using this parameter name throughout the code.